src/openMarket/user_interface/product/list_products/Container.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import React, { Component } from 'react';
+import { Link } from 'react-router';
+import ReactTable from 'react-table';
+import * as Rx from "rxjs";
+
+const columns = [{
+  header: 'Barcode',
+  accessor: 'barcode',
+  hideFilter: true
+},
+{
+  header: 'Name',
+  accessor: 'name',
+  filterMethod: (filter, row) => (row[filter.id].includes(filter.value))
+},
+{
+  header: 'Price',
+  accessor: 'price',
+  hideFilter: true
+},
+
+  {
+    header: 'Stock',
+    accessor: 'stock',
+    hideFilter: true
+  },
+  {
+    header: 'Category',
+    accessor: 'categoryId',
+    hideFilter: true
+  },
+  {
+    header: 'Status',
+    accessor: 'status',
+    hideFilter: true
+  }
+];
+const filterSubject$ = new Rx.Subject();
+
+class Container extends Component {
+
+  filterAndSorting(state, instance){
+    filterSubject$.next(state.filtering);
+  };
+
+  componentWillMount() {
+    const { listProductFetch, listProductFetchWithFilters } = this.props;
+
+    filterSubject$
+      .debounceTime(300)
+      .flatMap(filter => Rx.Observable.from(filter)
+        .map(filter => listProductFetchWithFilters(filter.value))
+        .defaultIfEmpty(Rx.Observable.of(listProductFetch()))
```

Comment:
because that's the exact behaviour that i wanna represent, if the user type something in the searchbox and after the filtering delete all text, will produce a empty search filter that means (for me) the user wanna delete the current filter and see again the listing without filters ;)
Keep in mind that there is no apply filters button...